### PR TITLE
Fix Don't show groups in share placeholder if group sharing is disabled #2242

### LINKED ISSUE #2242: Don't show groups in share placeholder if group sharing is disabled
### Steps to reproduce
1. Check out nextcloud/server#25658
2. Disallow sharing with groups in the sharing settings
3. Open the share section on an address book

### Expected behaviour
No mention of "groups" in the placeholder.

### Actual behaviour
Mention of "groups" in the placeholder.

### PR DESCRIPTION
**Pull request by Zishan-7**
Fix #2242 by checking group sharing enabled or not then updating the value of placeholder based on that value

### CODE REVIEW

File: lib/Controller/PageController.php [hunk 1]
```diff
@@ -106,6 +106,8 @@ public function index(): TemplateResponse {
 		$isCirclesEnabled = $this->appManager->isEnabledForUser('circles') === true;
 		// if circles is not installed, we use 0.0.0
 		$isCircleVersionCompatible = $this->compareVersion->isCompatible($circleVersion ? $circleVersion : '0.0.0', 22);
+		// Check whether group sharing is enabled or not
+		$isGroupSharingEnabled = $this->config->getAppValue('core', 'shareapi_allow_group_sharing', 'yes') === 'yes';
```
Comment (skjnldsv) [inline lib/Controller/PageController.php:110]: You also need to provide it through the initial state

```php
$this->initialStateService->provideInitialState(Application::APP_ID, 'isGroupSharingEnabled', $isGroupSharingEnabled);
```

Comment (Zishan-7) [reply lib/Controller/PageController.php:110]: Thanks for the suggestion, I have added it.

**Review by tcitworld — COMMENTED** [review]
LGTM. Can you [squash](https://levelup.gitconnected.com/how-to-squash-git-commits-9a095c1bc1fc) your commits?

**Review by ChristophWurst — APPROVED** [review]
Looks good!

**Comment by hamza221** [comment]
Fixed lint check, and rebased.

**Pull request merged**